gwpy/timeseries/timeseries.py

```diff
@@ -1388,28 +1388,143 @@ def demodulate(self, f, stride=1, exp=False, deg=True):
+        evolution, rather than a fixed frequency.
+
+        Unlike :meth:`~gwpy.timeseries.TimeSeries.demodulate`, the complex
+        output is double-sided by default, so is not multiplied by 2.
+
+        Examples
+        --------
+        Heterodyning can be useful in examining quasi-monochromatic signals
+        with a known phase evolution. Examples of such signals are the
+        `continuous-wave hardware injections
+        <https://www.gw-openscience.org/s6hwcw/>`_ in many of the LIGO
+        observing runs, which are designed to mimic the expected signals
+        from rapidly rotating neutron stars. For these sources the signal's
+        frequency slowly decreases and is also Doppler modulated due to
+        the Earth's rotational and orbital motion.
+
+        We can show an example of demodulation by creating a simulated signal
+        with a phase evolution described by a second order Taylor expansion in
+        phase, i.e., it is defined by a frequency and first frequency
+        derivative. We can download some LIGO Livingston data from GWOSC and
+        inject such a signal and attempt to recover the amplitude.
+
+        >>> from gwpy.timeseries import TimeSeries, TimeSeriesDict
+        >>> data = TimeSeries.fetch_open_data('L1', 1131350417, 1131357617)
+
+        We now need to set the signal parameters, generate the expected
+        phase evolution, and create the signal:
+
+        >>> f0 = 123.456789  # signal frequency (Hz)
+        >>> fdot = -9.87654321e-7  # signal frequency derivative (Hz/s)
+        >>> fpeoch = 1131350417  # phase epoch
+        >>> amp = 1.5e-22  # signal amplitude
+        >>> phase0 = 0.4  # signal phase at the phase epoch
+        >>> import numpy
+        >>> times = data.times.value - fepoch
+        >>> phase = 2 * numpy.pi * (f0 * times + 0.5 * fdot * times**2)
+        >>> signal = TimeSeries(amp * numpy.cos(phase + phase0),
+        >>>                     sample_rate=data.sample_rate, t0=data.t0)
+        >>> data = data.inject(signal)
+
+        We now bandpass the data around the signal frequency, and then
+        heterodyne the `TimeSeries` using this phase evolution with a stride of
+        one minute:
+
+        >>> filtdata = data.bandpass(f0 - 0.5, f0 + 0.5)
+        >>> het = filtdata.heterodyne(phase, stride=60, singlesided=True)
+
+        Let's also demodulate the signal at a fixed freqeuncy for comparison
+
+        >>> demod = filtdata.demodulate(f0, stride=60, exp=True)
+
+        We can then plot signal amplitude over time (cropping the first two
+        minutes to remove the filter response):
+
+        >>> combined = TimeSeriesDict()
+        >>> combined['Heterodyned'] = het.crop(het.x0.value + 180).abs()
+        >>> combined['Demodulated'] = demod.crop(demod.x0.value + 180).abs()
+        >>> plot = combined.plot()
+        >>> ax = plot.gca()
+        >>> ax.legend()
+        >>> ax.set_ylabel("Signal Strain Amplitude")
```

Comment:
For consistency with other gwpy examples:

```suggestion
        >>> ax.set_ylabel("Strain amplitude")
```

Reply:
Done.